source/blog/2018-03-02-containers-w-skopeo-ostree-oci.md

```diff
@@ -0,0 +1,283 @@
+- adjust `namespaces`: to add `{"type": "network"}` to make a separated network stack otherwise it would use host networking
+- you can adjust mapping between users `"linux": { "uidMappings": [ ... ] }` typically containers root is the current user
+
+here is how you can attach a writable directory for `/data` (which is `cont1/data` we have created before)
+
+```
+{
+// ...
+	"mounts": [
+		{
+			"destination": "/data",
+			"type": "bind",
+			"source": "data",
+			"options": ["rbind","rw"]
+		},
+    // ...
+    ]
+// ...
+}
+```
+
+To run the container type `runc run` followed by any name like `redis`
+
+```
+$ runc run redis 
+1:C 03 Mar 16:13:06.463 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
+1:C 03 Mar 16:13:06.474 # Redis version=4.0.8, bits=64, commit=00000000, modified=0, pid=1, just started
+...                _._                                                  
+           _.-``__ ''-._                                             
+      _.-``    `.  `_.  ''-._           Redis 4.0.8 (00000000/0) 64 bit
+  .-`` .-```.  ```\/    _.,_ ''-._                                   
+ (    '      ,       .-`  | `,    )     Running in standalone mode
+ |`-._`-...-` __...-.``-._|'` _.-'|     Port: 6379
+ |    `-._   `._    /     _.-'    |     PID: 1
+  `-._    `-._  `-./  _.-'    _.-'                                   
+ |`-._`-._    `-.__.-'    _.-'_.-'|                                  
+ |    `-._`-._        _.-'_.-'    |           http://redis.io        
+  `-._    `-._`-.__.-'_.-'    _.-'                                   
+ |`-._`-._    `-.__.-'    _.-'_.-'|                                  
+ |    `-._`-._        _.-'_.-'    |                                  
+  `-._    `-._`-.__.-'_.-'    _.-'                                   
+      `-._    `-.__.-'    _.-'                                       
+          `-._        _.-'                                           
+              `-.__.-'                                               
+
+```
+
+In another terminal you can have a shell inside the container using `runc exec redis /bin/sh`
+
+```
+$ runc exec redis /bin/sh
+/data # ps -a
+PID   USER     TIME   COMMAND
+    1 root       0:00 redis-server
+   18 root       0:00 /bin/sh
+   24 root       0:00 ps -a
+/data # 
+```
+
+### Bubble Wraps OCI
+
+`bwrap-oci` is another OCI runtime that is designed for userspace containers (non-root)
+You can use the same `config.json` we made in previous section.
+There was [a bug](https://github.com/projectatomic/bwrap-oci/pull/17) in `bwrap-oci`,
+that's why you need to build it from source.
+
+```
+$ bwrap-oci run redis
+```
+
+You can list running Bubble wrapped containers using `bwrap-oci list`
+
+```
+$ bwrap-oci list
+NAME                          PID       STATUS    BUNDLE
+redis                         23369     running   /home/alsadi/ostree/cont1
+```
+
+Unfortunately there is no `bwrap-oci exec`, 
+
```

Comment:
can you add a section explaining that this is what "atomic install --system|--user" does behind the scenes and show the command for doing it?

Reply:
done